### PR TITLE
 Reverse Alternate k nodes in a linked list in Java

### LINKED ISSUE #2538: Reverse alternate k nodes in a Singly Linked List in Java
## 🚀 Feature

I want to implement the solution to reverse alternate k nodes in a singly linked list in Java as a part of GSSOC'21.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by Meghna180401**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

This contains the solution to reverse alternate k nodes in a singly linked list in Java.

### Checklist

- [✅ ] I've read the contribution guidelines.
- [✅ ] I've checked the issue list before deciding what to submit.
- [✅ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #2538

### CODE REVIEW

**Comment by Meghna180401** [comment]
@HarshCasper @bhav09 Kindly review.

File: Java/ds/Reverse_alternate_k_Nodes.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+Algorithm:
+1.Reverse the first k nodes
+2.Link the next of head of the modified list to the (k+1)th node 
+3.Traverse the next k nodes to skip their reversal
+4.Since 2k nodes are already processed,recursively call the function for the rest n-2k nodes
+5.Return the new head of the list
+
+Time Complexity : O(n)
+Space Complexity : O(n),since there is a recursive function
```
Comment (HarshCasper) [inline Java/ds/Reverse_alternate_k_Nodes.java]: Remove these

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Closing this PR due to inactivity.

**Comment by Meghna180401** [comment]
@ankitaggarwal23 Due to a death in my family, I was inactive. I can now dedicate the required time to this.
So I request you to kindly re-open this issue as I have my code prepared with the changes requested.

**Comment by ankitaggarwal23** [comment]
Okay I can understand, Sure I reopened this you can start working!

**Comment by Meghna180401** [comment]
Thanks a lot @ankitaggarwal23 .
I have updated the code with required changes. Kindly review.

File: Java/ds/Reverse_alternate_k_Nodes.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
+﻿//Given the head pointer to a linked list of length n and an integer k, we have to reverse every alternate k nodes in the linked list.
+
```
Comment (ankitaggarwal23) [inline Java/ds/Reverse_alternate_k_Nodes.java]: Add a bit more meaningful description

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: Java/ds/Reverse_alternate_k_Nodes.java (hunk 2 above)
Comment (Meghna180401) [reply Java/ds/Reverse_alternate_k_Nodes.java]: @ankitaggarwal23 Should I add a sample input/output in the description to make it more comprehensible ?

Comment (ankitaggarwal23) [reply Java/ds/Reverse_alternate_k_Nodes.java]: No you can add your approach or may be if you used any algo you can add a bit more about algo etc.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**